python-package/securexgboost/remote_server.py

```diff
@@ -210,20 +212,23 @@ def rpc_XGBoosterSetParam(self, request, context):
         Set booster parameter
         """
         try:
-            _ = self._synchronize(remote_api.XGBoosterSetParam, request)
-            return remote_pb2.Status(status=0)
+            sig, sig_len = self._synchronize(remote_api.XGBoosterSetParam, request)
+            sig_proto = pointer_to_proto(sig, sig_len)
```

Comment:
This line seems to be repeated in various places. Can we call `pointer_to_proto(sig, sig_len)` before returning out of `self._synchronize`? Maybe we can call it in `invoke()` after running `self._func` before setting `self._ret`?

Reply:
The trouble is that `self._synchronize` simply returns whatever is returned by the invoked function, and the number of values returned by each function are different. So, `invoke` has no knowledge of which returned value is the signature.